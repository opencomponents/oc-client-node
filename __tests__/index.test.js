'use strict';

const client = require('../src');
jest.mock('../src/components-renderer');
jest.mock('../src/get-components-info');
jest.mock('../src/warmup');

test('typeof client to equal "function"', () => {
  expect(typeof client).toEqual('function');
});

describe('client api', () => {
  const api = client();

  test('to match snapshot', () => {
    expect(api).toMatchSnapshot();
  });

  test('getComponentsInfo callback to have been called', () => {
    const callback = jest.fn();
    api.getComponentsInfo(null, callback);
    expect(callback).toHaveBeenCalled();
  });

  test('init callback to have been called', () => {
    const callback = jest.fn();
    api.init(null, callback);
    expect(callback).toHaveBeenCalled();
  });

  test('renderComponent (w/ 2 params) callback to have been called', () => {
    const callback = jest.fn();
    api.renderComponent('good', callback);
    expect(callback).toHaveBeenCalled();
  });

  test('renderComponent (w/ 3 params) callback to have been called', () => {
    const callback = jest.fn();
    api.renderComponent('bad', {}, callback);
    expect(callback).toHaveBeenCalled();
  });

  test('renderComponents (w/ 2 params) callback to have been called', () => {
    const callback = jest.fn();
    api.renderComponents([{ name: 'bad' }], callback);
    expect(callback).toHaveBeenCalled();
  });

  test('renderComponents (w/ 3 params) callback to have been called', () => {
    const callback = jest.fn();
    api.renderComponents([{ name: 'good' }], {}, callback);
    expect(callback).toHaveBeenCalled();
  });
});

test('error', () => {
  expect(() => {
    client({ registries: {} });
  }).toThrowErrorMatchingSnapshot();
});
