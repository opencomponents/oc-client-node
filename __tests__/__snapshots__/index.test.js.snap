// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client api to match snapshot 1`] = `
Object {
  "getComponentsInfo": [Function],
  "init": [Function],
  "renderComponent": [Function],
  "renderComponents": [Function],
  "renderTemplate": [Function],
}
`;

exports[`error 1`] = `"Configuration is not valid: registries must contain at least one endpoint"`;
